sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            MonitorBase telemetry = new MonitorBase();
+
+            var telemetryType = activity.GetTelemetryType();
+            telemetry.BaseType = Telemetry_Base_Type_Mapping[telemetryType];
+            string url = ProcessHttpTags(activity.Tags);
+
+            if (telemetryType == TelemetryType.Request)
+            {
+                var request = new RequestData(2, activity.Context.SpanId.ToHexString(), activity.Duration.ToString("c", CultureInfo.InvariantCulture), activity.GetStatus().IsOk, activity.GetStatusCode())
+                {
+                    Name = activity.DisplayName,
+                    Url = url,
+                    // TODO: Handle request.source.
+                };
+
+                // TODO: Handle activity.TagObjects
+                ExtractPropertiesFromTags(request.Properties, activity.Tags);
+
+                telemetry.BaseData = request;
+            }
+            else if (telemetryType == TelemetryType.Dependency)
+            {
+                var dependency = new RemoteDependencyData(2, activity.DisplayName, activity.Duration)
+                {
+                    Id = activity.Context.SpanId.ToHexString(),
+                    Success = activity.GetStatus().IsOk
+                };
+
+                // TODO: Handle activity.TagObjects
+                ExtractPropertiesFromTags(dependency.Properties, activity.Tags);
+
+                if (url != null)
+                {
+                    dependency.Data = url;
+                    dependency.Type = "HTTP"; // TODO: Parse for storage / SB.
+                    dependency.ResultCode = activity.GetStatusCode();
+                }
+
+                // TODO: Handle dependency.target.
+                telemetry.BaseData = dependency;
+            }
+
+            return telemetry;
+        }
+
+        private static string ProcessHttpTags(IEnumerable<KeyValuePair<string, string>> tags)
+        {
+            var httpTags = tags.Where(item => item.Key.StartsWith("http.", StringComparison.InvariantCulture))
+                               .ToDictionary(item => item.Key, item => item.Value);
+
+
+            httpTags.TryGetValue(SemanticConventions.AttributeHttpUrl, out var url);
+            if (url != null)
+            {
+                return url;
+            }
+
+            httpTags.TryGetValue(SemanticConventions.AttributeHttpScheme, out var httpScheme);
+            httpTags.TryGetValue(SemanticConventions.AttributeHttpHost, out var httpHost);
+            httpTags.TryGetValue(SemanticConventions.AttributeHttpTarget, out var httpTarget);
+
+            if (httpHost != null)
+            {
+                url = httpScheme + httpHost  + httpTarget;
+                return url;
+            }
+
+            // TODO: Follow spec - https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#http-client
+
+            return url;
```

Comment:
should this be `null` here? At line 144, you're only returning URL if it's != null.

Reply:
Yes. This will return `null`, this part needs to be revisited to align with OT Spec.